solr/core/src/java/org/apache/solr/request/json/RequestUtil.java

```diff
@@ -191,7 +191,9 @@ public static void processParams(SolrRequestHandler handler, SolrQueryRequest re
     }
 
     // implement compat for existing components...
-    JsonQueryConverter jsonQueryConverter = new JsonQueryConverter();
+    JsonQueryConverter jsonQueryConverter = (JsonQueryConverter) req.getContext()
```

Comment:
I think it's necessary only for the earlier "dynamic" approach. I suppose it's not necessary in the eager one.

Reply:
True. Reversed.